File: mobile/PhoneHub/app/src/main/java/com/phonehub/android/models/SilentModeSession.kt
package com.phonehub.android.models

import android.os.Parcelable
import kotlinx.parcelize.Parcelize
import java.util.*

/**
 * Gizli Mod Oturumu Modeli
 */
@Parcelize
data class SilentModeSession(
    val id: String = UUID.randomUUID().toString(),
    val deviceId: String = "",
    val userId: String = "",
    val startedAt: Long = 0,
    val endedAt: Long? = null,
    val operationCount: Int = 0,
    val operations: MutableList<String> = mutableListOf(),
    val ipAddress: String? = null
) : Parcelable

File: backend/PhoneHub.Core/Models/SilentModeSession.cs
namespace PhoneHub.Core.Models;

/// <summary>
/// Gizli Mod Oturumu - Gizli işlemleri izlemek için
/// </summary>
public class SilentModeSession
{
    public Guid Id { get; set; }
    public Guid DeviceId { get; set; }
    public Guid UserId { get; set; }

    /// <summary>
    /// Oturum başlangıcı
    /// </summary>
    public DateTime StartedAt { get; set; }

    /// <summary>
    /// Oturum bitişi
    /// </summary>
    public DateTime? EndedAt { get; set; }

    /// <summary>
    /// Gizli işlem sayısı
    /// </summary>
    public int OperationCount { get; set; }

    /// <summary>
    /// Yapılan işlemler
    /// Örnek: "Uygulama açıldı", "Dosya silinit", "SMS gönderildi"
    /// </summary>
    public List<string>? Operations { get; set; } = new();

    /// <summary>
    /// İşlemin dökmesi - JSON olarak
    /// </summary>
    public string? OperationLogs { get; set; }

    /// <summary>
    /// Gizli mod sırasında ekran durumu kaydı
    /// </summary>
    public string? DisplayStateHistory { get; set; }

    /// <summary>
    /// IP Adresi
    /// </summary>
    public string? IPAddress { get; set; }

    public User? User { get; set; }
    public Device? Device { get; set; }
}
